pkg/sql/alter_table.go

```diff
@@ -1831,7 +1831,7 @@ func dropColumnImpl(
 		}
 	}
 	if !found {
-		return nil, pgerror.Newf(pgcode.ObjectNotInPrerequisiteState,
+		return nil, pgerror.Newf(pgcode.FeatureNotSupported,
```

Comment:
This is to please `TestCompareLegacyAndDeclarative` in which we assert that if some DDL returns an error in legacy schema changer but succeeds in declarative schema changer, the error must be FeatureNotSupported error.